### PR TITLE
[bitnami/etcd] Disable metrics even if podAnnotations are provided

### LINKED ISSUE #4145: [bitnami/etcd] If podAnnotations are provided prometheus annotations are injected even if metrics are disabled.
**Which chart**:
ETCD 4.12.X

**Describe the bug**
If a podAnnotation is provided the the check for metrics activation is skipped and prometheus podAnnotations are provided anyway.

**To Reproduce**

values.yaml
```yaml
podAnnotations:
  test: "test"

metrics:
  enabled: false
```
```
helm template etcd-test . -f ./values-test.yaml --dry-run
```

the templating result in StatefulSet is  :
```yaml
  template:
    metadata:
      labels:
        app.kubernetes.io/name: etcd
        helm.sh/chart: etcd-4.12.1
        app.kubernetes.io/instance: etcd-test
        app.kubernetes.io/managed-by: Helm
      annotations:
        test: test
        prometheus.io/port: "2379"
        prometheus.io/scrape: "true"
```

**Expected behavior**
If `metrics.enabled` is set to `false` metrics should be disabled
or if `metrics.serviceMonitor.enabled` is set to `true` prometheus annotations should not be injected.

```yaml
podAnnotations:
  test: "test"
```
or 
```yaml
podAnnotations:
  test: "test"

metrics:
  enabled: true
  serviceMonitor:
    enabled: true
```

should give 
```yaml
  template:
    metadata:
      labels:
        app.kubernetes.io/name: etcd
        helm.sh/chart: etcd-4.12.1
        app.kubernetes.io/instance: etcd-test
        app.kubernetes.io/managed-by: Helm
      annotations:
        test: test
```

**Additional context**
Opening a PR to fix that.

### PR DESCRIPTION
**Pull request by GeckoSplinter**
**Description of the change**

Add test for metrics activation before templating the podAnnotations related to prometheus scraping.

**Benefits**

Now you can add generic podAnnotations without activating prometheus scraping annotations.
Or if we activate serviceMonitor the prometheus annotations are not enabled.

**Possible drawbacks**

**Applicable issues**

  - fixes #4145

**Additional information**

**Checklist** 
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)
- [x] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files

:warning: Keep in mind that if you want to make changes to the kubeapps chart, please implement them in the [kubeapps repository](https://github.com/kubeapps/kubeapps/tree/master/chart/kubeapps). This is only a synchronized mirror.

### CODE REVIEW

File: bitnami/etcd/templates/statefulset.yaml [hunk 1]
```diff
@@ -25,9 +25,11 @@ spec:
         {{- if .Values.podAnnotations }}
         {{- include "etcd.tplValue" ( dict "value" .Values.podAnnotations "context" $) | nindent 8 }}
         {{- end }}
-        {{- if .Values.metrics.podAnnotations }}
+        {{- if and .Values.metrics.enabled .Values.metrics.podAnnotations }}
+        {{- if not (and .Values.metrics.serviceMonitor .Values.metrics.serviceMonitor.enabled) }}
```
Comment (andresbono) [inline bitnami/etcd/templates/statefulset.yaml:29]: Wouldn't it be the same than the following?
```suggestion
        {{- if and .Values.metrics.enabled .Values.metrics.podAnnotations ( not .Values.metrics.serviceMonitor.enabled ) }}
```

If you change this, you should remove the `{{ - end }}` below.

Comment (andresbono) [reply bitnami/etcd/templates/statefulset.yaml:29]: Line 23 should be updated as well, right? There could be situations in which the annotations would be empty, i.e.:

- `metrics.enabled=true`
- `metrics.serviceMonitor.enabled=true`

**Comment by andresbono** [comment]
Thanks for your contribution!! Could you please check my comments?

File: bitnami/etcd/templates/statefulset.yaml (hunk 1 above)
Comment (GeckoSplinter) [reply bitnami/etcd/templates/statefulset.yaml:29]: Yes indeed if miss that the `metrics.serviceMonitor` is present in the default values.
And yes line 23 should be updated

File: bitnami/etcd/templates/statefulset.yaml [hunk 2]
```diff
@@ -20,12 +20,12 @@ spec:
   template:
     metadata:
       labels: {{- include "etcd.labels" . | nindent 8 }}
-      {{- if or .Values.podAnnotations .Values.metrics.enabled }}
+      {{- if or .Values.podAnnotations (and .Values.metrics.enabled (not .Values.metrics.serviceMonitor.enabled)) }}
```
Comment (andresbono) [inline bitnami/etcd/templates/statefulset.yaml:23]: I think this is also necessary to be consistent with the other two conditions below:

```suggestion
      {{- if or .Values.podAnnotations (and .Values.metrics.enabled .Values.metrics.podAnnotations (not .Values.metrics.serviceMonitor.enabled)) }}
```

**Review by andresbono — APPROVED** [review]
LGTM

**Comment by andresbono** [comment]
Thanks for your contribution and for addressing the suggestions, @GeckoSplinter!!

**Pull request merged**